src/core/ext/transport/chaotic_good/frame.cc

```diff
@@ -115,7 +120,11 @@ absl::StatusOr<Arena::PoolPtr<Metadata>> ReadMetadata(
     uint32_t stream_id, bool is_header, bool is_client) {
   if (!maybe_slices.ok()) return maybe_slices.status();
   auto& slices = *maybe_slices;
-  Arena::PoolPtr<Metadata> metadata;
+  MemoryAllocator memory_allocator = MemoryAllocator(
+      ResourceQuota::Default()->memory_quota()->CreateMemoryAllocator(
+          "read_metadata"));
+  auto arena = MakeScopedArena(1024, &memory_allocator);
```

Comment:
I dont think we should be allocating an arena each time a frame is parsed. We should change the ServerFragmentFrame::Deserialize and ReadMetadata methods to plumb the arena ptr from the transport down to this spot.

Reply:
Yeah, thanks for catching this! I will create arena in one place of the transport and pass the ptr down.